pkg/frontend/transport/handler_test.go

```diff
@@ -97,3 +98,62 @@ func TestHandler_ServeHTTP(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_FailedRoundTrip(t *testing.T) {
+	for _, test := range []struct {
+		name                string
+		expectedMetrics     int
+		path                string
+		expectQueryParamLog bool
+		queryErr            error
+	}{
+		{
+			name:                "Failed round trip with context cancelled",
+			expectedMetrics:     0,
+			path:                "/api/v1/query?query=up&time=2015-07-01T20:10:51.781Z",
+			expectQueryParamLog: true,
+			queryErr:            context.Canceled,
+		},
+		{
+			name:                "Failed round trip with no query params",
+			expectedMetrics:     0,
+			path:                "/api/v1/query",
+			expectQueryParamLog: false,
+			queryErr:            context.Canceled,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			roundTripper := roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+				return nil, test.queryErr
+			})
+
+			reg := prometheus.NewPedanticRegistry()
+			logs := &concurrency.SyncBuffer{}
+			logger := log.NewLogfmtLogger(logs)
+			handler := NewHandler(HandlerConfig{QueryStatsEnabled: true}, roundTripper, logger, reg)
```

Comment:
You don't need `QueryStatsEnabled: true`, since the tested code doesn't use it. I think specifying it is misleading, since it makes you think the test depends on it.

```suggestion
			handler := NewHandler(HandlerConfig{}, roundTripper, logger, reg)
```

Reply:
Now that we will be using `reportQueryStats` it will be required, otherwise `stats` is not correctly initialized, which leads to changes needing to be made to `reportQueryStats` - I wanted to avoid polluting this function with a bunch of `if` statements if possible

Reply:
I don't see any problem using `{QueryStatsEnabled: true}`.

Reply:
The problem is not all tenants will have this config set to True, so will need to modify `reportQueryStats` to not report stats here: https://github.com/grafana/mimir/blob/main/pkg/frontend/transport/handler.go#L203-L208

Reply:
If we are going to use `reportQueryStats`, `QueryStatsEnabled` should be relevant, so my comment is stale.